### DISCUSSION TITLE
cloudimage

### DISCUSSION THREAD

**Discussion by tinect** [FriendsOfShopware/FroshPlatformThumbnailProcessor#39] [Patterns]
This pattern is for [cloudimage](https://www.cloudimage.io/en/home)
My opinion to this service: has free plan, fast and including webp

```
https://TOKEN.cloudimg.io/www.myshop.com/{mediaPath}?w={width}&func=bound&org_if_sml=1
```
becomes `https://TOKEN.cloudimg.io/www.myshop.com/media/01/01/01/tinect.png?w=800&func=bound&org_if_sml=1`

Replace `TOKEN` and `www.myshop.com`.
You can set up aliases instead of your `www.myshop.com`. Do not forget to set up whitelisted domains.

Hints:
- set whitelisted domains within cloudimage OR set your url as Origin prefix so you can remove it from your pattern
- activate compression for SVG at cloudimage + add SVG as valid file extension for compression in plugin configuration

**Comment by ghost** [comment]
Just a little typo: `?` is needed after {mediaPath}:

`https://token.cloudimg.io/v7/www.example.com/{mediaPath}?w={width}&func=bound`

**Reply by tinect** [reply]
Oh yes. Thank you. I adjusted the sample.

**Comment by OrnaminMarketing** [comment]
Wanna point out that the v7 in the url has to be left out now, unless your account is from before November 2021.

**Reply by tinect** [reply]
Thank you. I've added the note and added parameter `org_if_sml=1`

**Comment by thesec** [comment]
How can the pattern be changed to automatically serve webp?

**Reply by thesec** [reply]
Cloudimage has an option in the settings that will automatically generate and serve webp.  
If needs be, image formats can be forced individually by adding the query parameter `force_format`; e.g. `force_format=webp`.

**Reply by tinect** [reply]
But they do serve it in webp automatically, regarding their docs:
https://docs.cloudimage.io/go/cloudimage-documentation-v7/en/image-compression

> By default, Cloudimage delivers the images as WebP. If the format is not supported by the client browser, it falls back to JPEG or PNG.

My opinion: I would not set a forced format. What if they avif-support, but not every device supports it? Or jpegXL. Then you would have problems by having to stick to the forced format.

**Reply by thesec** [reply]
Am just listing options and not recommending anything :)

Cloudimage does have an option that needs to be ticked in their preferences to automatically convert to webp.
I have asked them for clarification on the discrepancy with their doc.

Also, they do not serve webp by default but only if the user agent is claiming to be capable rendering it ... unless the image format is forced and the handshake results are being ignored.

**Reply by thesec** [reply]
Just clarified with Cloudimage:
Webp is not automatically served in general but only for the formats selected in the according preference.
Formats not selected for conversion will be compressed only.

**Comment by cmichael-de** [comment]
Works great! Thx!

Can someone help with 2 questions:

1. SEO: The URL of an image starts with https://TOKEN.cloudimg.io now. Is this SEO relevant - for example is google saving these cloud urls in its index?
2. General Data Protection Regulation (DSGVO): The Image src contains an external url, is there a consent tool (europe dsgvo) required for cloudimages?

**Reply by tinect** [reply]
Hi, nice to hear this. :-) 

Because these discussions here do just resch handful of people and the questions are more general than plugin related, 
you might want to check in forums or slack for answers.
